aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -798,16 +800,72 @@ Tensor select_sparse_csr(const Tensor& self, int64_t dim, int64_t index) {
         options.pinned_memory_opt());
   } else if (dim < n_batch + 2) {
     // Selecting sparse dimension
-    TORCH_CHECK(
-        self.layout() == kSparseCsr || self.layout() == kSparseCsc,
-        "select(): selecting non-batch dimensions is currently only supported for non-blocked sparse compressed layouts tensors.");
     TORCH_CHECK(
         n_batch == 0,
-        "select(): selecting rows or columns is not implemented for batched sparse compressed tensors.")
-    // Converting to COO and calling select is slightly slower than operating
-    // on the CSR indices directly for constructing a COO vector, however
-    // current version is more readable and easier to understand.
-    return self.to_sparse().select(dim, index);
+        "select(): selecting sparse dimensions is not implemented for batched sparse compressed tensors.")
+    TORCH_INTERNAL_ASSERT(dim == 0 || dim == 1);
+    DimVector blocksize{1, 1};
```

Comment:
Given that here we select over sparse dimensions conditioned on `n_batches == 0`, can we extend that to arbitrary value of `n_batches`?